bookings/api.py

```diff
@@ -24,6 +25,20 @@ class BookingViewSet(
     def get_queryset(self):
         return Booking.objects.for_maas_operator(self.request.user.maas_operator)
 
+    def retrieve(self, request, *args, **kwargs):
+        booking = self.get_object()
+
+        passthrough_parameters = PassthroughParametersSerializer(
+            data=request.query_params
+        )
+        passthrough_parameters.is_valid(raise_exception=True)
+        tickets = booking.retrieve(passthrough_parameters.validated_data)
+
+        booking_data = self.serializer_class(instance=booking).data
+        booking_data["tickets"] = tickets
+
+        return Response(booking_data)
```

Comment:
If you want, I think you could replace some of this code by adding `RetrieveModelMixin` to the viewset.

Reply:
It's quite sleek like this too. Shouldn't tickets be added anyway using passthrough_parameters? This way all the data is immediately visible